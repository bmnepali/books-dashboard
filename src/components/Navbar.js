import React from 'react';

const BOOK_LOGO = '../assets/images/logo.png';
const USER_AVATAR = '../assets/images/user.png';

const Navbar = () => {
  return (
    <nav className="navbar fixed-top flex-md-nowrap p-0">
      <span className="mobile-menu d-md-none">
        <i className="icon ion-md-menu" />
      </span>
      <a className="navbar-brand col-sm-3 col-md-2 mr-0" href="#">
        <img src={BOOK_LOGO} className="logo" alt="Books Dashboard" />
      </a>
      <span className="mobile-profile-setting d-md-none">
        <i className="icon ion-md-more" />
      </span>
      <div className="navbar-menu-content w-100">
        <div className="search-form-wrapper w-50">
          <form className="search-books">
            <input className="form-control" type="text" placeholder="Search" aria-label="Search" />
            <button className="btn btn-link" type="submit">
              <i className="icon ion-md-search" />
            </button>
          </form>
        </div>
        <ul className="navbar-nav navbar-custom">
          <li className="nav-item notifications">
            <span data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
              <i className="icon ion-md-notifications-outline" />
              <sup>1</sup>
            </span>
            <div className="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownMenuButton">
              <a className="dropdown-item" href="#">
                Notification One
              </a>
              <a className="dropdown-item" href="#">
                Notification two
              </a>
              <a className="dropdown-item" href="#">
                Notification three
              </a>
            </div>
          </li>
          <li className="nav-item user-info-block">
            <div className="media" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
              <div className="media-body">
                <h5 className="mt-0 mb-1">Brawn Smith</h5>
              </div>
              <img src={USER_AVATAR} className="align-self-center ml-3" alt="Brawn Smith" />
            </div>
            <div className="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownMenuButton">
              <a className="dropdown-item" href="#">
                Profile
              </a>
              <a className="dropdown-item" href="#">
                Mail
              </a>
              <a className="dropdown-item" href="#">
                Logout
              </a>
            </div>
          </li>
        </ul>
      </div>
    </nav>
  );
};

export default Navbar;
