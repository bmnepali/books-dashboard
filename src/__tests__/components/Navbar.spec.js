import * as React from 'react';
import Navbar from '../../components/Navbar';

describe('Navbar Component', () => {
  it('renders the component without any error', () => {
    shallow(<Navbar />);
  });

  it('matches the snapshot', () => {
    expect(toJson(shallow(<Navbar />))).toMatchSnapshot();
  });

  describe('checking navbar component closely', () => {
    let component = null;

    beforeEach(() => {
      component = mount(<Navbar />);
    });

    it('renders the logo correctly', () => {
      expect(component.find('.logo').length).toBe(1);
      expect(component.find('.logo').getElement().props.src).toEqual('../assets/images/logo.png');
    });

    it('renders the user avatar correctly', () => {
      expect(component.find('.align-self-center').length).toBe(1);
      expect(component.find('.align-self-center').getElement().props.src).toEqual('../assets/images/user.png');
    });

    it('renders search books from correctly', () => {
      expect(component.find('.search-books').length).toBe(1);
      expect(component.find('input').getElement().props.placeholder).toEqual('Search');
      expect(component.find('button').getElement().props.type).toEqual('submit');
    });
  });
});

