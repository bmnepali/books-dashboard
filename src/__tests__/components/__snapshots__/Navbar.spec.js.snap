// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches the snapshot 1`] = `
<nav
  className="navbar fixed-top flex-md-nowrap p-0"
>
  <span
    className="mobile-menu d-md-none"
  >
    <i
      className="icon ion-md-menu"
    />
  </span>
  <a
    className="navbar-brand col-sm-3 col-md-2 mr-0"
    href="#"
  >
    <img
      alt="Books Dashboard"
      className="logo"
      src="../assets/images/logo.png"
    />
  </a>
  <span
    className="mobile-profile-setting d-md-none"
  >
    <i
      className="icon ion-md-more"
    />
  </span>
  <div
    className="navbar-menu-content w-100"
  >
    <div
      className="search-form-wrapper w-50"
    >
      <form
        className="search-books"
      >
        <input
          aria-label="Search"
          className="form-control"
          placeholder="Search"
          type="text"
        />
        <button
          className="btn btn-link"
          type="submit"
        >
          <i
            className="icon ion-md-search"
          />
        </button>
      </form>
    </div>
    <ul
      className="navbar-nav navbar-custom"
    >
      <li
        className="nav-item notifications"
      >
        <span
          aria-expanded="false"
          aria-haspopup="true"
          data-toggle="dropdown"
        >
          <i
            className="icon ion-md-notifications-outline"
          />
          <sup>
            1
          </sup>
        </span>
        <div
          aria-labelledby="dropdownMenuButton"
          className="dropdown-menu dropdown-menu-right"
        >
          <a
            className="dropdown-item"
            href="#"
          >
            Notification One
          </a>
          <a
            className="dropdown-item"
            href="#"
          >
            Notification two
          </a>
          <a
            className="dropdown-item"
            href="#"
          >
            Notification three
          </a>
        </div>
      </li>
      <li
        className="nav-item user-info-block"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          className="media"
          data-toggle="dropdown"
        >
          <div
            className="media-body"
          >
            <h5
              className="mt-0 mb-1"
            >
              Brawn Smith
            </h5>
          </div>
          <img
            alt="Brawn Smith"
            className="align-self-center ml-3"
            src="../assets/images/user.png"
          />
        </div>
        <div
          aria-labelledby="dropdownMenuButton"
          className="dropdown-menu dropdown-menu-right"
        >
          <a
            className="dropdown-item"
            href="#"
          >
            Profile
          </a>
          <a
            className="dropdown-item"
            href="#"
          >
            Mail
          </a>
          <a
            className="dropdown-item"
            href="#"
          >
            Logout
          </a>
        </div>
      </li>
    </ul>
  </div>
</nav>
`;
